### PR TITLE
sys+cpu: Some warning fixes

### PR DESCRIPTION
**Pull request by gebart**
Fixes some of the warnings enabled by -Wextra

References #1121

### CODE REVIEW

**Comment by gebart** [comment]
(Murdock was idle)

File: cpu/cortexm_common/cortexm_init.c [hunk 1]
```diff
@@ -46,7 +46,7 @@ void cortexm_init(void)
     /* set SVC interrupt to same priority as the rest */
     NVIC_SetPriority(SVCall_IRQn, CPU_DEFAULT_IRQ_PRIO);
     /* initialize all vendor specific interrupts with the same value */
-    for (int i = 0; i < CPU_IRQ_NUMOF; i++) {
+    for (int i = 0; i < (int)CPU_IRQ_NUMOF; i++) {
```
Comment (aabadie) [inline cpu/cortexm_common/cortexm_init.c]: Instead of the cast, could `unsigned` be used like in the other places ?

Comment (gebart) [inline cpu/cortexm_common/cortexm_init.c]: I guess, but IRQn_Type is a signed enum below, so I wasn't sure what would be the best solution here.

Comment (miri64) [inline cpu/cortexm_common/cortexm_init.c]: (*without much knowledge about that particular type*) why are you starting to count at 0 then?

Comment (miri64) [inline cpu/cortexm_common/cortexm_init.c]: Or use the type directly for counting?

Comment (LudwigKnuepfer) [inline cpu/cortexm_common/cortexm_init.c]: | why are you starting to count at 0 then?
That is what has been done before ..

**Comment by PeterKietzmann** [comment]
Murdock was happy but merge conflict in file auto_init_si70xx.c

File: cpu/cortexm_common/cortexm_init.c (hunk 1 above)
Comment (gebart) [inline cpu/cortexm_common/cortexm_init.c]: The vendor specific interrupts start at number 0, which is what is being initialized here, but there are 16 core interrupts which have negative indices.

**Comment by gebart** [comment]
Rebased

**Comment by miri64** [comment]
Jenkins once again is failing in qemu => unrelated

**Pull request merged**